core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -327,6 +357,96 @@ public long handleWayTags(ReaderWay way, long allowed, long relationFlags) {
         }
     }
 
+    private long extractMeter(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
+        String value = extractRoadAttribute(way, keys);
+
+        if (Helper.isEmpty(value)) return flags;
+
+        double val;
+        try {
+            val = stringToMeter(value);
+        } catch (Throwable t) {
+            LOG.warn("Unable to extract meter from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
+            val = 0.0;
+        }
+
+        return valueEncoder.setDoubleValue(flags, val);
+    }
+
+    private long extractTons(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
+        String value = extractRoadAttribute(way, keys);
+
+        if (Helper.isEmpty(value)) return flags;
+
+        double val;
+        try {
+            val = stringToTons(value);
+        } catch (Throwable t) {
+            LOG.warn("Unable to extract tons from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
+            val = 0.0;
+        }
+
+        return valueEncoder.setDoubleValue(flags, val);
+    }
+
+    private String extractRoadAttribute(ReaderWay way, List<String> keys) {
```

Comment:
let us keep it here for now

> Also if two several keys exist, only the last one is returned. Not sure if this is an issue

I think this is not a problem yet. (BTW: the first vs. last?)